### PR TITLE
Ensure notify when proxy connections disconnect

### PR DESCRIPTION
**Pull request by dnhatn**
TransportService doesn't respond to the pending requests of proxy connections when the underlying connections get disconnected because proxy connections do not override the [getCacheKey](https://github.com/elastic/elasticsearch/blob/1c3ddf8ff1199b11d498fe2d9ef70eb65b505855/server/src/main/java/org/elasticsearch/transport/TransportService.java#L970-L971) method. Some CCS requests would never be completed because of this bug.

I found this while working on the cross-clusters task cancellation in #65443.

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/search/ccs/CrossClusterSearchIT.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins(String clusterAlias) {
+        if (clusterAlias.equals(LOCAL_CLUSTER)) {
+            return super.nodePlugins(clusterAlias);
+        } else {
+            final List<Class<? extends Plugin>> plugins = new ArrayList<>(super.nodePlugins(clusterAlias));
+            plugins.add(SearchListenerPlugin.class);
+            return plugins;
```
Comment (original-brownbear) [inline server/src/internalClusterTest/java/org/elasticsearch/search/ccs/CrossClusterSearchIT.java:138]: NIT: we have `CollectionUtils.appendToCopy(super.nodePlugins(clusterAlias), SearchListenerPlugin.class);` to make this stuff one line

File: server/src/internalClusterTest/java/org/elasticsearch/search/ccs/CrossClusterSearchIT.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        @Override
+        public void onIndexModule(IndexModule indexModule) {
+            indexModule.addSearchOperationListener(new SearchOperationListener() {
+                @Override
+                public void onPreQueryPhase(SearchContext searchContext) {
+                    startedLatch.get().countDown();
+                    final CountDownLatch latch = queryLatch.get();
+                    if (latch != null) {
+                        try {
+                            latch.await();
```
Comment (original-brownbear) [inline server/src/internalClusterTest/java/org/elasticsearch/search/ccs/CrossClusterSearchIT.java:179]: NIT: Maybe put a timeout on these `.await` here and above and `assertTrue` to make test failures a little more obvious?

**Review by original-brownbear — APPROVED** [review]
LGTM

File: server/src/internalClusterTest/java/org/elasticsearch/search/ccs/CrossClusterSearchIT.java (hunk 1 above)
Comment (dnhatn) [reply server/src/internalClusterTest/java/org/elasticsearch/search/ccs/CrossClusterSearchIT.java]: Nice, see https://github.com/elastic/elasticsearch/pull/65697/commits/3e262a7349647bb6b3c48ab1b840d88396aa08ce

File: server/src/internalClusterTest/java/org/elasticsearch/search/ccs/CrossClusterSearchIT.java (hunk 2 above)
Comment (dnhatn) [reply server/src/internalClusterTest/java/org/elasticsearch/search/ccs/CrossClusterSearchIT.java]: Sure, adjusted in https://github.com/elastic/elasticsearch/pull/65697/commits/031bf821152fb07cc37e7ff63b89d298aa491ae3

**Comment by dnhatn** [comment]
Thanks Armin!

**Pull request merged**